tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/DynamoDb/AwsDynamoDbCommon.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="AwsDynamoDbCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.DynamoDb
+{
+    internal static class AwsDynamoDbCommon
+    {
+        private const string DatadogAwsDynamoDbServiceName = "aws-dynamodb";
+        private const string DynamoDbServiceName = "DynamoDB";
+        private const string DynamoDbOperationName = "aws.dynamodb.request";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AwsDynamoDbCommon));
+
+        internal const string IntegrationName = nameof(Configuration.IntegrationId.AwsDynamoDb);
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.AwsDynamoDb;
+
+        public static Scope CreateScope(Tracer tracer, string operation, out AwsDynamoDbTags tags, ISpanContext parentContext = null)
+        {
+            tags = null;
+
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId) || !tracer.Settings.IsIntegrationEnabled(AwsConstants.IntegrationId))
```

Comment:
Do we need to do this here? I thought if the setting turned the integration off then it won't be instrumented therefore not reaching here?

Reply:
I can't answer this question, maybe @zacharycmontoya has the knowledge around it.

Either way, this seemed like a common pattern for integrations.

Reply:
Yes we do need this here. We automatically set up all of the bytecode instrumentations regardless if they're enabled or not. But when our handlers are executed we check the configuration to see if we should create spans